google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -1640,6 +1650,7 @@ ApiFuture<QuerySnapshot> get(@Nullable ByteString transactionId) {
     internalStream(
         new QuerySnapshotObserver() {
           final List<QueryDocumentSnapshot> documentSnapshots = new ArrayList<>();
+          boolean hasReturned = false;
```

Comment:
Maybe rename to `hasCompleted`.

Also add comment: // The stream's onCompleted could be called more than once, this flag makes sure only the first one is actually processed.